### PR TITLE
LinkResolver refactor

### PR DESCRIPTION
**Pull request by mangas**

### CODE REVIEW

**Review by lutter — APPROVED** [review]
LGTM. It would be good to have a commit message that explains why this change is necessary

File: tests/Cargo.toml [hunk 1]
```diff
@@ -4,10 +4,16 @@ version = "0.25.2"
...
 tokio-stream = "0.1"
 futures = "0.3.13"
 port_check = "0.1.5"
 anyhow = "1.0.54"
 lazy_static = "1.4.0"
+graph-node = { path = "../node" }
+graph-core = { path = "../core" }
+graph = { path = "../graph" }
+graph-mock = { path = "../mock" }
+graph-store-postgres = { path = "../store/postgres" }
```
Comment (leoyvens) [inline tests/Cargo.toml:19]: Why did all these deps become necessary?

Comment (mangas) [reply tests/Cargo.toml:19]: This is for code that's not yet committed, the tests I've been working on

Comment (leoyvens) [reply tests/Cargo.toml:19]: Then they should be removed if they are dead deps for now.

Comment (leoyvens) [reply tests/Cargo.toml:19]: Then they should be removed if they are dead deps for now.

File: core/src/subgraph/instance_manager.rs [hunk 2]
```diff
@@ -154,7 +144,7 @@ where
                 deployment.hash.cheap_clone(),
                 manifest,
                 // Allow for infinite retries for subgraph definition files.
-                &self.link_resolver.as_ref().clone().with_retries(),
```
Comment (leoyvens) [inline core/src/subgraph/instance_manager.rs:157]: Was the `with_retries()` lost in the refactor?

Comment (mangas) [reply core/src/subgraph/instance_manager.rs:157]: https://github.com/graphprotocol/graph-node/pull/3447

**Pull request merged**